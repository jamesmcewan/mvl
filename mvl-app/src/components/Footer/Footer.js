import React from 'react';

const Footer = () => (
  <footer>
    <p>{`Â© Marvel`}</p>
  </footer>
);

export default Footer;
