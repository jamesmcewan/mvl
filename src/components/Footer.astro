---
import MightyDinosaur from './mighty-dinosaur.astro'
const { attributionText } = Astro.props
---

<footer class="bg-rose-900">
  <div class="mx-auto max-w-7xl p-2 sm:p-6 lg:p-8">
    <p class="font-montserrat font-light text-rose-100">
      {attributionText}
      {`Comics are Â© ${new Date().getFullYear()} Marvel`}
    </p>
    <div class="w-40 py-2">
      <MightyDinosaur />
    </div>
  </div>
</footer>
