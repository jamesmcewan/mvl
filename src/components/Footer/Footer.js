import React from 'react';
import styled from 'styled-components';

const StyledFooter = styled.footer`
  text-align: center;
  margin-bottom: 120px;
`;

const Footer = ({ attributionText }) => (
  <StyledFooter>
    <p>{attributionText}</p>
  </StyledFooter>
);

export default Footer;
