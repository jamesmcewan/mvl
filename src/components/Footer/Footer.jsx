const Footer = () => (
  <footer>
    <p>{`Â© Marvel`}</p>
  </footer>
)

export default Footer
