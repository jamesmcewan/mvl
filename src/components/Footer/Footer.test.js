import React from 'react';
import { mount } from 'enzyme';
import Footer from './Footer';
import 'jest-styled-components';

describe('Footer', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Footer />);
    expect(wrapper).toMatchSnapshot();
  });
});
