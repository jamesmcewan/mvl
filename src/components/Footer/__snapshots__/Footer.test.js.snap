// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the correct JSX 1`] = `
.c0 {
  text-align: center;
  margin-bottom: 120px;
}

<Footer>
  <styled.footer>
    <footer
      className="c0"
    >
      <p />
    </footer>
  </styled.footer>
</Footer>
`;
